src/storage/knn_index/knn_diskann/inner/diskann_utils.cppm

```diff
@@ -0,0 +1,481 @@
+// Copyright(C) 2023 InfiniFlow, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module;
+
+
+#include <cstdlib>
+#include <boost/dynamic_bitset.hpp>
+
+import stl;
+import third_party;
+import default_values;
+import infinity_exception;
+import file_system;
+
+export module diskann_utils;
+
+namespace infinity {
+export void FileHandlerSeek(FileHandler &fh, i64 offset) {
+    auto &fs = fh.file_system_;
+    fs.Seek(fh, offset);
+}
+
+export inline void AllocAligned(void **ptr, SizeT size, SizeT align){
+    *ptr = nullptr;
+    if ((u64)size % (u64)align != 0){
+        UnrecoverableError(fmt::format("AllocAligned(): size {} is not aligned to {}", size, align));
+    }
+
+    *ptr = std::aligned_alloc(align, size);
+    if (*ptr == nullptr){
+        UnrecoverableError("AllocAligned(): Failed to allocate memory");
+    }
+}
+
+export inline void AlignedFree(void *ptr){
+    if (ptr == nullptr)
+        return;
+
+    free(ptr);
+}
+
+export struct Neighbor {
+    u32 id;
+    f32 distance;
+    bool expanded;
+
+    Neighbor() = default;
+
+    Neighbor(u32 id, f32 distance) : id(id), distance(distance), expanded(false) {}
+
+    inline bool operator<(const Neighbor &other) const {
+        return distance < other.distance || (distance == other.distance && id < other.id);
+    }
+
+    inline bool operator==(const Neighbor &other) const {
+        return (id == other.id);
+    }
+
+};
+
```

Comment:
Expanded nodes in the Neighbor queue still need to be saved, and priority_queue doesn't support random visits, it can only keep popping to find unexpanded nodes making the use of priority_queue require additional data structures and operations.

Reply:
std::make_heap supports random access.